resources/aws_ec2.go

```diff
@@ -454,6 +497,61 @@ func (obj *AwsEc2Res) longpollWatch() error {
 	}
 }
 
+// snsWatch uses amazon cloudwatch events and simple notification service to
+// detect ec2 instance state changes.
+func (obj *AwsEc2Res) snsWatch() error {
+	send := false
+	var exit *error
+	var err error
+	// make sns client
+	snsClient, err := obj.snsClient()
+	if err != nil {
+		return errwrap.Wrapf(err, "error making sns client")
+	}
+	// start the endpoint
+	obj.wg.Add(1)
+	go func() {
+		defer obj.wg.Done()
+		log.Printf("%s: Starting SNS Endpoint", obj)
+		if err := obj.snsEndpoint.ListenAndServe(); err != nil {
+			select {
+			case obj.awsChan <- &chanStruct{
+				err: errwrap.Wrapf(err, "sns server error"),
+			}:
+			case <-obj.closeChan:
+			}
+		}
+	}()
+	// subscribe endpoint to the topic
+	if err = obj.snsSubscribe(snsClient); err != nil {
+		return errwrap.Wrapf(err, "error subscribing to sns topic")
+	}
+	// process events
+	for {
+		select {
+		case event := <-obj.Events():
+			if exit, send = obj.ReadEvent(event); exit != nil {
+				close(obj.closeChan)
```

Comment:
I noticed this, and then wondered if we had made this possible mistake before, and I think this is yes, so I opened https://github.com/purpleidea/mgmt/issues/245